src/main/java/com/zaxxer/hikari/pool/PoolEntry.java

```diff
@@ -96,9 +108,21 @@ public void setKeepalive(ScheduledFuture<?> keepalive) {
       this.keepalive = keepalive;
    }
 
-   Connection createProxyConnection(final ProxyLeakTask leakTask)
-   {
-      return ProxyFactory.getProxyConnection(this, connection, openStatements, leakTask, isReadOnly, isAutoCommit);
+   Connection createProxyConnection(final ProxyLeakTask leakTask){
+      Connection newproxyconn= ProxyFactory.getProxyConnection(this, connection, openStatements, leakTask, isReadOnly, isAutoCommit);
+      // calling beginRequest for JDBC 4.3 and later
```

Comment:
I think we can mention issue# here in comment for a better explanation of the changes.